example/qds_spark_example.py

```diff
@@ -0,0 +1,115 @@
+        content = content_file.read()
+    return content
+
+
+def execute_query(cluster_label, cmd_to_run, language, user_program_arguments=None, arguments=None):
+    """
+    Helper method to execute a script
+    :param cluster_label:
+    :param cmd_to_run:
+    :param language:
+    :param user_program_arguments:
+    :param arguments:
+    :return:
+    """
+    if script is None or script == "":
+        print("script cannot be None or empty")
+        return None
+
+    if not language:
+        print("language cannot be None or empty")
+        return
+
+    if language in ["command_line"]:
+        # A Shell command needs to be invoked in this fashion
+        cmd = SparkCommand.create(label=cluster_label, cmdline=cmd_to_run, arguments=arguments,
+                                  user_program_arguments=user_program_arguments)
+    elif language == "sql":
+        # A SQL command needs to be invoked in this fashion
+        cmd = SparkCommand.create(label=cluster_label, sql=cmd_to_run, arguments=arguments)
+    else:
+        # A python, R or scala command needs to be invoked in this fashion.
+        cmd = SparkCommand.create(label=cluster_label, program=cmd_to_run, language=language,
+                                  arguments=arguments, user_program_arguments=user_program_arguments)
+
+    while not SparkCommand.is_done(cmd.status):
+        print("Waiting for completion of command : {}".format(cmd.id))
+        cmd = SparkCommand.find(cmd.id)
+        time.sleep(5)
+
+    if SparkCommand.is_success(cmd.status):
+        print("\nCommand Executed: Completed successfully")
+    else:
+        print("\nCommand Executed: Failed!!!. The status returned is: {}".format(cmd.status))
+        print(cmd.get_log())
+    return cmd
+
+
+def get_results(command):
+    """
+    A helper method to get the results
+    :param command:
+    :return:
+    """
+    if command is None:
+        return None
+
+    results_file_name = get_results_filename(command.id)
+    fp = open(results_file_name, 'w')
+
+    command.get_results(fp, delim="\n")
+    print("results are written to {}".format(results_file_name))
+
+
+if __name__ == '__main__':
+    # Set the API token. If you are using any other environment other then api.qubole.com then set api_url to that url
+    # as <env_url>/api
+    Qubole.configure(api_token='<auth_token>')
+
+    filename = "<your script location>"
+    user_program_arguments = None # arguments for your script
+    arguments = None # spark configuration for your program for ex : "--conf spark.executor.memory=1024M"
+    cluster_label = "<your cluster label>" # the cluster on which the command will run
+
+    # Running a python command. In case of scala or R get the script content and then set the langauage field to scala
+    # or R as required
+    script = get_content(filename)
+    command = execute_query(cluster_label, script, "python", user_program_arguments=user_program_arguments,
```

Comment:
Can we break it down to different examples and files:

* python program (pyspark) with Spark command. Input specified as file
* python program (pyspark) with Spark command. Input inlined as code
* Spark SQL command 
* Spark submit command line submission

Reply:
Implemented the below
1. pyspark program with a file maintained externally
2. scala program with the code inline
3. on SQL command
4. One spark submit command line